service/src/main/java/bio/terra/workspace/service/workspace/CloudSyncRoleMapping.java

```diff
@@ -110,23 +113,91 @@ public class CloudSyncRoleMapping {
-  public static final ImmutableMap<WsmIamRole, CustomGcpIamRole> CUSTOM_GCP_PROJECT_IAM_ROLES =
-      ImmutableMap.of(
-          WsmIamRole.OWNER, PROJECT_OWNER,
-          // TODO: this should map to PROJECT_APPLICATION if that's created.
-          WsmIamRole.APPLICATION, PROJECT_WRITER,
-          WsmIamRole.WRITER, PROJECT_WRITER,
-          WsmIamRole.READER, PROJECT_READER);
+  private static final List<String> PROJECT_READER_DATAPROC_PERMISSIONS =
+      new ImmutableList.Builder<String>()
+          .addAll(PROJECT_READER_PERMISSIONS)
+          .add(
+              "dataproc.clusters.get",
+              "dataproc.clusters.list",
+              "dataproc.jobs.get",
+              "dataproc.jobs.list")
+          .build();
+
+  private static final List<String> PROJECT_OWNER_DATAPROC_PERMISSIONS =
+      new ImmutableList.Builder<String>()
+          .addAll(PROJECT_READER_DATAPROC_PERMISSIONS)
+          .addAll(PROJECT_OWNER_PERMISSIONS)
+          .add(
+              // Dataproc CRUD permissions
+              // TODO: Move read permissions to PROJECT_READER and revoke modify permissions once
+              // adding wsm managed dataproc
+              "dataproc.clusters.create",
+              "dataproc.clusters.update",
+              "dataproc.clusters.delete",
+              "dataproc.clusters.use",
+              "dataproc.clusters.start",
+              "dataproc.clusters.stop",
+              "dataproc.jobs.create",
+              "dataproc.jobs.update",
+              "dataproc.jobs.delete",
+              "dataproc.jobs.cancel",
+              "dataproc.autoscalingPolicies.get",
+              "dataproc.autoscalingPolicies.list",
+              "dataproc.autoscalingPolicies.create",
+              "dataproc.autoscalingPolicies.update",
+              "dataproc.autoscalingPolicies.delete",
+              // TODO: Remove all of the following once adding WSM managed dataproc
+              "dataproc.tasks.lease",
+              "dataproc.tasks.listInvalidatedLeases",
+              "dataproc.tasks.reportStatus",
+              "dataproc.agents.get",
+              "dataproc.agents.create",
+              "dataproc.agents.update",
+              "dataproc.agents.delete",
+              "logging.logEntries.create",
+              "monitoring.metricDescriptors.create",
+              "monitoring.metricDescriptors.get",
+              "monitoring.metricDescriptors.list",
+              "monitoring.monitoredResourceDescriptors.get",
+              "monitoring.monitoredResourceDescriptors.list",
+              "monitoring.timeSeries.create",
+              "compute.machineTypes.get",
+              "compute.networks.get",
+              "compute.networks.list",
+              "compute.projects.get",
+              "compute.regions.get",
+              "compute.regions.list",
+              "compute.zones.get",
+              "compute.zones.list")
+          .build();
+
+  public CloudSyncRoleMapping() {
```

Comment:
I don't think this will work as-is. FeatureConfiguration is a Spring component (because it's annotated with `@Configuration` so the FeatureConfiguration singleton will have values from `application.yml`, but a newly-created FeatureConfiguration object (like here) won't have anything.
There are two approaches you could take here:

1. Change this constructor to take a FeatureConfiguration object as a parameter, and have all callers pass in the FeatureConfiguration bean. Inside a flight you can get this from the `FlightBeanBag`, and you can then pass it to steps as needed

2. Make `CloudSyncRoleMapping` a Spring component as well. Then, you can add a FeatureConfiguration parameter to the constructor and use `@Autowired` to ensure this is wired up to the FeatureConfiguration bean. To call this component's methods inside flights, you would add `CloudSyncRoleMapping` to FlightBeanBag and then pass the object from that to all steps as needed

Either approach should work, but I'd probably lean towards 2 a bit

Reply:
Ah thanks! Flight bean bag was the missing piece.

Reply:
+1 for 2.